appservice/main.py

```diff
@@ -666,14 +672,16 @@ def process_message(self, message: discord.Message) -> Tuple[str, Dict]:
                     f"<@{char}{member.id}>", f"@{member.username}"
                 )
 
-        # `except_deleted` for invalid channels.
-        # TODO can this block for too long ?
-        for channel in re.findall(f"<#([0-9]{{{discord.ID_LEN}}})>", content):
-            discord_channel = except_deleted(self.get_channel)(channel)
-            name = (
-                discord_channel.name if discord_channel else "deleted-channel"
-            )
-            content = content.replace(f"<#{channel}>", f"#{name}")
+        # Replace channel IDs with names.
+        channels = re.findall("<#([0-9]+)>", content)
+        if channels:
+            discord_channels = self.get_channels(message.guild_id)
```

Comment:
Could you add something like this, just in-case?

```
if not message.guild_id:
    self.logger.warning(f"Message '{message.id}' in channel '{message.channel_id}' does not contain a guild_id!")
else:
    # Do mention stuff
```

Reply:
Added, I also added a comment in Message class.